refurbish/tests/Main.hs

```diff
@@ -82,90 +84,154 @@ instance TO.IsOption VerboseOutput where
+            [ T.testGroup "Rewriting" $ sweeten e <$> rewriteTests
+            , T.testGroup "Injection" $ sweeten e <$> injectionTests
+            ]
   where
-    ingredients = T.includingOptions [ TO.Option (Proxy @UseDockerRunner)
-                                     , TO.Option (Proxy @VerboseOutput)
-                                     ] : T.defaultIngredients
-
-mkTests :: C.HandleAllocator -> [FilePath] -> IO (Logger, Executor) -> T.TestTree
-mkTests hdlAlloc exes executor =
-      let callTest :: (IO (Logger, Executor) -> C.HandleAllocator -> R.LayoutStrategy -> a -> T.TestTree)
-                   -> R.LayoutStrategy -> a -> T.TestTree
-          callTest f = f executor hdlAlloc
-          injection = [ ("tests/injection-base/injection-base.ppc64.exe"
-                        , "tests/injection-base/ppc64-exit.bin")
-                      ]
-      in T.testGroup "RefurbishTests" [
-        callTest rewritingTests (R.LayoutStrategy R.Parallel R.BlockGrouping R.AlwaysTrampoline) exes,
-        callTest rewritingTests (R.LayoutStrategy R.Parallel R.BlockGrouping R.WholeFunctionTrampoline) exes,
-        callTest rewritingTests (R.LayoutStrategy (R.Compact R.SortedOrder) R.BlockGrouping R.AlwaysTrampoline) exes,
-        callTest rewritingTests (R.LayoutStrategy (R.Compact R.SortedOrder) R.LoopGrouping R.AlwaysTrampoline) exes,
-        callTest rewritingTests (R.LayoutStrategy (R.Compact R.SortedOrder) R.FunctionGrouping R.WholeFunctionTrampoline) exes,
-        callTest codeInjectionTests (R.LayoutStrategy R.Parallel R.BlockGrouping R.AlwaysTrampoline) injection,
-        callTest codeInjectionTests (R.LayoutStrategy (R.Compact R.SortedOrder) R.BlockGrouping R.AlwaysTrampoline) injection
-        ]
+    ingredients = T.includingOptions (TO.Option (Proxy @UseDockerRunner)
+                                      : TO.Option (Proxy @VerboseOutput)
+                                       : TS.sugarOptions)
+                  : TS.sugarIngredients [rewriteCube, injectionCube]
+                  <> T.defaultIngredients
+
+----------------------------------------------------------------------
+
+data Sweetener = Sweetener (IO (Logger, Executor) -> T.TestTree)
+               | Sachet String [Sweetener]
+
+sweeten :: IO (Logger, Executor) -> Sweetener -> T.TestTree
+sweeten mkExecutor (Sweetener s) = s mkExecutor
+sweeten mkExecutor (Sachet nm s) = T.testGroup nm (sweeten mkExecutor <$> s)
+
+strategies :: [R.LayoutStrategy]
+strategies = [ R.LayoutStrategy allocator grouping trampoline
+             | allocator <- [ R.Parallel
+                            , R.Compact R.SortedOrder
+                            ]
+             , grouping <- [ R.BlockGrouping
+                           , R.LoopGrouping
+                           , R.FunctionGrouping
+                           ]
+             , trampoline <- [ R.WholeFunctionTrampoline
+                             , R.AlwaysTrampoline
+                             ]
+             ]
```

Comment:
Do we have a path forward if we want to not always do a full matrix (e.g., just try a few tests in the `FunctionGrouping` mode, but a full set of tests in `'Parallel`)?

Reply:
My thought was to add support for the `CI_TEST_LEVEL` env var; in the `mkRewriteTests` and `mkInjectionTests` we can add logic to check this or other determiners and return `[]` to suppress tests.